src/modules/timepix3/readout/DataParser.cpp

```diff
@@ -0,0 +1,84 @@
+// Copyright (C) 2023 European Spallation Source, ERIC. See LICENSE file
+//===----------------------------------------------------------------------===//
+///
+/// \file
+///
+/// \brief Parser for ESS readout of Timepix3 Modules
+//===----------------------------------------------------------------------===//
+
+#include <timepix3/readout/DataParser.h>
+#include <common/debug/Trace.h>
+#include <common/readout/ess/Parser.h>
+
+#undef TRC_LEVEL
+#define TRC_LEVEL TRC_L_DEB
+
+namespace Timepix3 {
+
+// Assume we start after the PacketHeader
+int DataParser::parse(const char *Buffer, unsigned int Size) {
+  Result.clear();
+  unsigned int ParsedReadouts = 0;
+
+  unsigned int BytesLeft = Size;
+  char *DataPtr = (char *)Buffer;
+
+  while (BytesLeft) {
+    uint64_t dataBytes;
+
+    if (BytesLeft <= sizeof(dataBytes)){
+      //TODO add some error handling here
+    }
+    // Copy 2 bytes starting from offset 0 into dcolBytes variable
+    memcpy(&dataBytes, DataPtr, sizeof(dataBytes));
+
+    uint8_t packet_type = (dataBytes & 0xF000000000000000) >> 60;
+    if (packet_type == 11){
+      Timepix3PixelReadout Data;
+
+      Data.dcol = (dataBytes & 0x0FE0000000000000) >> 52;
```

Comment:
It is probably more intuitive to shift the data first and then AND with short numbers.

Reply:
I like being able to see all the masks one after the other, and see that where dcol ends, ie an E, lines up with where spix starts, ie a 1. Personally that makes more sense to me.